### PR TITLE
Reduce Float usage

### PR DESCRIPTION
**Pull request by dkubb**
* This branch replaces Float with BigDecimal for more accurate calculations
  when possible.

### CODE REVIEW

File: .rspec [hunk 1]
```diff
@@ -1 +1,3 @@
 --color
+--require spec_helper
+#--format progress
```
Comment (dkubb) [inline .rspec]: @mbj I put it there to remind me to look and see why the specs are busted.

I'll probably amend the commit so the line isn't present, and then have a separate commit that adds the line back uncommented and fixes the specs at once.

**Comment by dkubb** [comment]
@mbj FYI I realize we don't want to touch `spree_active_shipping`, but so much depends on it I wanted to do as quick a pass as possible to make sure the math it's doing is correct. I am not really refactoring any code or anything like that, simply doing the smallest possible transformations I can to make sure the math side of things is correct.

File: spec/models/weight_limits_spec.rb [hunk 2]
```diff
@@ -13,20 +14,20 @@ def build_content_items(variant, quantity, order)
 
     let(:country) { mock_model Spree::Country, :iso => "CA", :state_name => "Quebec", :state => nil }
     let(:address) { mock_model(Spree::Address, city: 'Montreal', state: build(:state, country: country), zipcode: 'H2B', country: country) }
-    let(:usa) { FactoryGirl.create(:country, :name => "USA", :iso => "US") }
-    let(:state) { FactoryGirl.create(:state, country: usa, abbr: 'MD', name: 'Maryland')}
-    let(:us_address) { FactoryGirl.create(:address, :country => usa, :state => state, :city => "Chevy Chase", :zipcode => "20815") }
+    let(:usa) { FactoryGirl.build(:country, :name => "USA", :iso => "US") }
```
Comment (dkubb) [inline spec/models/weight_limits_spec.rb]: Oh, I touched this line and missed changing it to ruby 1.9 style.

File: app/models/spree/calculator/shipping/active_shipping/base.rb [hunk 3]
```diff
@@ -85,10 +85,10 @@ def compute_package(package)
           rate = self.class.rate_from(rates_result)
 
           return nil unless rate
-          rate = rate.to_f + (Spree::ActiveShipping::Config[:handling_fee].to_f || 0.0)
+          rate = BigDecimal.new(rate) + (Spree::ActiveShipping::Config[:handling_fee] || 0)
```
Comment (mbj) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: We've probably tried this before, but `Config.fetch` with a 2nd default param does not exist?

OT: We dropped these handling fees anyway, so we could remove the entire calculation here in another iteration.

File: lib/spree/active_shipping/ups_override.rb [hunk 4]
```diff
@@ -62,8 +62,8 @@ def build_rate_request(origin, destination, packages, options={})
                          units << XmlNode.new("Code", imperial ? 'IN' : 'CM')
                        end
                        [:length,:width,:height].each do |axis|
-                         value = ((imperial ? package.inches(axis) : package.cm(axis)).to_f*1000).round/1000.0 # 3 decimals
-                         dimensions << XmlNode.new(axis.to_s.capitalize, "%.3f" % [value,0.1].max)
+                         value = BigDecimal.new(imperial ? package.inches(axis) : package.cm(axis)).round(3)
```
Comment (mbj) [inline lib/spree/active_shipping/ups_override.rb]: OT: The size calculation does not have any effect for our app, as the rates are negotiated to ignore size, also the dimensions are not used in our app (and should probably be dropped).

Comment (mbj) [inline lib/spree/active_shipping/ups_override.rb]: I see `round(3)` be called many times, should we consider to extract the `3` into a constant?

File: app/models/spree/calculator/shipping/active_shipping/base.rb (hunk 3 above)
Comment (dkubb) [inline app/models/spree/calculator/shipping/active_shipping/base.rb]: > We've probably tried this before, but Config.fetch with a 2nd default param does not exist?

Yeah, I think we've looked for this a few times. I did not find anything when I examined the base class: https://github.com/MountainRoseHerbs/spree/blob/master/core/app/models/spree/preferences/configuration.rb#L41-L55

> We dropped these handling fees anyway, so we could remove the entire calculation here in another iteration.

Sure. I can do a follow-up after this is merged.

File: lib/spree/active_shipping/ups_override.rb (hunk 4 above)
Comment (dkubb) [inline lib/spree/active_shipping/ups_override.rb]: I'm cool with dropping it in a later iteration.

File: spec/models/weight_limits_spec.rb [hunk 5]
```diff
@@ -14,9 +14,9 @@ def build_content_items(variant, quantity, order)
 
     let(:country) { mock_model Spree::Country, :iso => "CA", :state_name => "Quebec", :state => nil }
     let(:address) { mock_model(Spree::Address, city: 'Montreal', state: build(:state, country: country), zipcode: 'H2B', country: country) }
-    let(:usa) { FactoryGirl.create(:country, :name => "USA", :iso => "US") }
-    let(:state) { FactoryGirl.create(:state, country: usa, abbr: 'MD', name: 'Maryland')}
-    let(:us_address) { FactoryGirl.create(:address, :country => usa, :state => state, :city => "Chevy Chase", :zipcode => "20815") }
+    let(:usa) { FactoryGirl.build(:country, name: 'USA', iso: 'US') }
+    let(:state) { FactoryGirl.build(:state, country: usa, abbr: 'MD', name: 'Maryland')}
+    let(:us_address) { FactoryGirl.build(:address, country: usa, state: state, city: 'Chevy Chase', zipcode: '20815') }
```
Comment (dkubb) [inline spec/models/weight_limits_spec.rb]: Heh, I was trying to avoid it for that reason, but I'll do it anyway.

File: lib/spree/active_shipping/ups_override.rb (hunk 4 above)
Comment (dkubb) [inline lib/spree/active_shipping/ups_override.rb]: > I see round(3) be called many times, should we consider to extract the 3 into a constant?

As far as I know, each element's rounding amount just happens to be `3` more by accident than intention. They could very well be different, since those values are not really related. I'd probably only want to use a constant if the were related.
